### DISCUSSION TITLE
Duplicate key error missing "message"

### DISCUSSION THREAD

**Discussion by Stubbs** [prisma/prisma#18098] [Q&A]
My app is throwing a unique key violation when calling create, I've captured this and passed it to my apps error handler via a call to next, but the error doesn't include a message to go with the error when turned into JSON.

```ts
    try {
      const user = await prisma.user.create({
        data: {
          first_name: first_name,
          email: email
        }
      });

      res.json(user);
    } catch (err) {
      next(err);
    }
```

error handler:
```ts
module.exports = function (err, req, res, next) {
  // set locals, only providing error in development
  res.locals.message = err.message;
  res.locals.error = req.app.get('env') === 'development' ? err : {};

  res.status(400).json(err);
}
```

and the error itself as output to the response.
```js
{
  code: 'P2002',
  clientVersion: '4.10.1',
  meta: { target: [ 'email' ] }
}
```

If I console log the `err` object I get a whole bunch more info, including the message, so I guess the toJSON() method of PrismaClientKnownRequestError isn't adding it?

**Comment by Stubbs** [comment]
Even if I add a "message" field to the object it doesn't get rendered in the response, but I am able to add other fields and they DO get rendered.

**Reply by nikolasburk** [reply]
Hey @Stubbs 👋 were you able to resolve this problem by now? Otherwise out Support team will try to look into this in the next days 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
